<launch>
    <!-- Robot description for MPC -->
    <arg name="task_file" default="$(find kinova_mpc_controllers)/params/task.info"/>
    <param name="robot_description_mpc"
           command="$(find xacro)/xacro --inorder $(find kinova_description_for_controllers)/urdf/kinova.urdf.xacro"/>
    <param name="task_file" value="$(arg task_file)"/>

    <!-- Load controllers -->
    <arg name="namespace" default="my_gen3"/>
    <arg name="simulation" default="false"/>
    <arg name="prefix" default="arm_"/>

    <rosparam command="load" file="$(find kinova_mpc_controllers)/params/common.yaml" subst_value="true"/>
    <rosparam command="load" file="$(find kinova_mpc_controllers)/params/gains.yaml" unless="$(arg simulation)" subst_value="true"/>
    <rosparam command="load" file="$(find kinova_mpc_controllers)/params/gains_sim.yaml" if="$(arg simulation)" subst_value="true"/>

    <group ns="$(arg namespace)">
        <rosparam file="$(find kinova_mpc_controllers)/params/ros_control.yaml" command="load" subst_value="true"/>
    </group>

    <!-- Interactive reference -->
    <node pkg="ocs2_mobile_manipulator_example" type="mobile_manipulator_target" name="mobile_manipulator_target"
          output="screen" args="$(arg task_file)"/>

    <arg name="test_admittance" default="false"/>
    <group if="$(arg test_admittance)">
        <include file="$(find sensor_tools)/launch/ft_sensor_standalone.launch"/>
    </group>


</launch>